### PR TITLE
Add unit tests for filesystem cache

### LINKED ISSUE #1842: Add test coverage for `Twig_Cache_Filesystem`
The only unit test covering some parts of `Twig_Cache_Filesystem` is `Twig_Tests_EnvironmentTest::testExtensionsAreNotInitializedWhenRenderingACompiledTemplate()`. Therefore, add unit tests for `Twig_Cache_Filesystem`.

### PR DESCRIPTION
**Pull request by znerol**
Fixes #1842

### CODE REVIEW

File: test/Twig/Tests/Cache/FilesystemTest.php [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+ * @coversDefaultClass Twig_Cache_Filesystem
+ */
+class Twig_Tests_Cache_FilesystemTest extends PHPUnit_Framework_TestCase
+{
+    private $nonce;
+    private $classname;
+    private $directory;
+    private $cache;
+
+    public function setUp() {
```
Comment (stof) [inline test/Twig/Tests/Cache/FilesystemTest.php]: should be protected

File: test/Twig/Tests/Cache/FilesystemTest.php [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                } else {
+                    unlink($filename);
+                }
+            }
+            rmdir($this->directory);
+        }
+    }
+
+    /**
+     * @covers ::load
```
Comment (stof) [inline test/Twig/Tests/Cache/FilesystemTest.php]: we don't use ``@covers`` in the Twig testsuite

File: test/Twig/Tests/Cache/FilesystemTest.php [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                    } else {
+                        unlink($filename);
+                    }
+                }
+            }
+            rmdir($this->directory);
+        }
+    }
+
+    public function testLoad() {
```
Comment (Tobion) [inline test/Twig/Tests/Cache/FilesystemTest.php]: braces on separate line

**Comment by fabpot** [comment]
Thank you @znerol.

**Pull request merged**